### ISSUE TITLE
[BUG] llm_params from GenerationOptions are not considered during generate_user_intent in generation.py when single_call is False

### ISSUE THREAD

**Issue by uvnikgupta** [NVIDIA/NeMo-Guardrails#505]
Following is the code for the case when <code>single_call</code> is <code>False</code>
ln 400 <code> generation.generate_user_intent()</code>: 

```python
# We make this call with temperature 0 to have it as deterministic as possible.
with llm_params(llm, temperature=self.config.lowest_temperature):
    result = await llm_call(llm, prompt)
```

Whereas below is the code for the case when <code>single_call</code> is <code>True</code>
ln 1164 to 1170 <code> generation.generate_intent_steps_message()</code>:
```python
generation_options: GenerationOptions = generation_options_var.get()
additional_params = {
	**((generation_options and generation_options.llm_params) or {}),
	"temperature": self.config.lowest_temperature,
}
with llm_params(llm, **additional_params):
	result = await llm_call(llm, prompt)
```
Please fix the <code>generate_user_intent()</code> to take into account the <code>llm_params</code> from <code>GenerationOptions</code> same as being done in <code>generate_intent_steps_message()</code>

**Comment by drazvan** [comment]
Thanks for reporting this, @uvnikgupta! This looks like an easy fix if you want to contribute. Otherwise, we'll try to fix this for 0.10.0.

**Comment by Shridhar2205** [comment]
Hi, I am trying to use Nemoguardrails for a very basic example (https://github.com/NVIDIA/NeMo-Guardrails/tree/a45b71af7d1777e6cf6fff70f82e1dafc0a71872/examples/bots/hello_world). However, I'm getting the following error,

WARNING:nemoguardrails.actions.action_dispatcher:Error while execution generate_user_intent: Unsupported data type
WARNING  Error while execution generate_user_intent: Unsupported data type

Is this error related to this bug, or is it due to other issues with AzureOpenAI+NemoGuardrails (https://github.com/NVIDIA/NeMo-Guardrails/issues/72 & https://github.com/NVIDIA/NeMo-Guardrails/issues/72) ?

My config.yml file looks like this:
models:
  - type: main
    engine: azure
    model: xxxxxxxxxx
    parameters:
      azure_endpoint: xxxxxxxxxx
      api_version: 2023-09-15-preview
      deployment_name: xxxxxxxxxx
      api_key: xxxxxxxxxx
      temperature: 0.7
